src/utils/branchHelpers.js

```diff
@@ -0,0 +1,89 @@
+const shelljs = require('shelljs');
+const assert = require('assert');
+
+const repoParentFolder = process.cwd() + '/../';
+
+module.exports = {
+  // Checkout branch
+  checkout: function(repo, branchName, verbose) {
+    console.log(`Checking out ${repo.name}/${branchName}.`);
+    const repoFolder = `${repoParentFolder}${repo.name}`;
+
+    const checkout = shelljs.exec(
+      `
+      cd ${repoFolder} &&
+      git co ${branchName}
+    `,
+      {silent: !verbose}
+    );
+    assert(
+      checkout.code === 0,
+      ` - Unable to checkout ${repo.name}/${branchName}: ${checkout.stderr}.`
+    );
+  },
+
+  // Fast forward to upstream head
+  fastForward: function(repo, branchName, verbose) {
+    console.log(`Pulling changes from remote ${repo.name}/${branchName}.`);
+    const repoFolder = `${repoParentFolder}${repo.name}`;
+
+    const statusShort = shelljs.exec(`cd ${repoFolder} && git status -s`, {
+      silent: !verbose,
+    });
+    assert(
+      !statusShort.stdout,
+      'Branch has un-committed changes and is not clean.'
+    );
+    const fetchAndStatus = shelljs.exec(
+      `cd ${repoFolder} && git fetch --tags && git status`,
+      {
+        silent: !verbose,
+      }
+    );
+    assert(
+      !fetchAndStatus.stdout.includes('have diverged'),
+      'Local and origin branches have diverged.  Please manually handle this merge.'
+    );
+    const pull = shelljs.exec(
+      `cd ${repoFolder} && git merge --ff-only origin/${branchName}`,
+      {
+        silent: !verbose,
+      }
+    );
+    assert(pull.code === 0, ` - ${pull.stderr}`);
+  },
+
+  make: function(repo, branchName, verbose = true) {
```

Comment:
Would be good if we can combine this with the makeBranch.js file somehow.

Reply:
Agreed!  I didn't delete _makeBranch.js_ since it is being used by _buildkite-docker-compose-plugin.mjs_ but I plan to refactor that to use _branchHelpers_.  It is effectively a copy / paste with minor changes to the console outputs.

Reply:
Sounds good. I think I would prefer individual helpers where possible, e.g. keep makeBranch.js but update the content. Also make `fastForwardBranch.js` and `checkoutBranch.js`. This will generally keep the code more maintainable (instead of one huge growing utility).

Reply:
Good point.  Thoughts on throwing them all in a folder together ('/src/utils/branch') to avoid too much clutter?

Reply:
Works for me 👍